optimizely/event/event_processor.py

```diff
@@ -0,0 +1,225 @@
+# Copyright 2019 Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import threading
+import time
+
+from datetime import timedelta
+from six.moves import queue
+
+from .user_event import UserEvent
+from .event_factory import EventFactory
+from optimizely import logger as _logging
+from optimizely.closeable import Closeable
+from optimizely.event_dispatcher import EventDispatcher as default_event_dispatcher
+from optimizely.helpers import validator
+
+ABC = abc.ABCMeta('ABC', (object,), {'__slots__': ()})
+
+
+class EventProcessor(ABC):
+  """ Class encapsulating event_processor functionality. Override with your own processor
+  providing process method. """
+
+  @abc.abstractmethod
+  def process(user_event):
+    pass
+
+
+class BatchEventProcessor(EventProcessor, Closeable):
```

Comment:
So here is what I would recommend. 

You should define a timer class that does stuff based on a defined interval. I am hoping that the timer class will be useful in both config manager and here in the batch event processor.

Reply:
do you mean something similar to this https://github.com/optimizely/ruby-sdk/blob/master/lib/optimizely/config_manager/async_scheduler.rb